src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -192,6 +192,28 @@ public void visit(NodeTraversal t, Node n, Node parent) {
         visitRequireCall(t, n, parent);
       }
 
+      // Finds calls to require.ensure which is a method to allow async loading of
+      // CommonJS modules:
+      //
+      // require.ensure(['/path/to/module1', '/path/to/module2'], function(require) {
+      //    var module1 = require('/path/to/module1');
+      //    var module2 = require('/path/to/module2');
+      // });
+      //
+      // will be rewritten as an IIFE
+      //
+      // (function() {
+      //   var module1 = require('/path/to/module1');
+      //   var module2 = require('/path/to/module2');
+      // })()
+      //
+      // See http://wiki.commonjs.org/wiki/Modules/Async/A
+      //     http://www.injectjs.com/docs/0.7.x/cjs/require.ensure.html
+      if (n.isCall() && n.getChildCount() == 3 && n.getFirstChild().isGetProp()
+          && n.getFirstChild().matchesQualifiedName("require.ensure")
+          && n.getSecondChild().isArrayLit() && n.getChildAtIndex(2).isFunction()) {
```

Comment:
Yeah, I'd prefer checking here up to `matchesQualifiedName()`, moving the rest to `visitRequireEnsureCall`, and updating the comment about supported pattern there.